### ISSUE TITLE
Implement logits softcap in paged attention

### ISSUE THREAD

**Issue by AleksKnezevic** [tenstorrent/tt-xla#4156]
gemma 4 required logits softcap i.e. `tanh(qk/cap) * cap` in attention before the softmax. This will require a change in the SDPA kernel to add optional cap and piping to get it back out to the vLLM plugin. 

This should be reasonably simple and we can try it ourselves before pulling in the ops team. If the change to the kernel is more complicated than a few hours, let's re-assess. Also, since it's a core (albeit optional) change to a performant kernel, you should get codeowner buy in (even over slack) before spending too much time on it.

**Comment by sshonTT** [comment]
I have updated a branch for softcap : https://github.com/tenstorrent/tt-xla/tree/sshon/pr-softcap-sdpa
Test Command : 
 - pytest -svv tests/torch/graphs/test_attention.py::test_gemma4_softcap_attention
 - pytest -svv tests/torch/ops/test_custom_ops.py::test_paged_scaled_dot_product_attention_decode_softcap
 - pytest -svv tests/torch/ops/test_custom_ops.py::test_scaled_dot_product_attention_decode_softcap
 - pytest -svv tt-xla/tests/torch/ops/test_custom_ops.py::test_scaled_dot_product_attention_softcap

This is just for a PoC without discussion with op team, waiting for feedback from them. (@pavlejosipovic)

**Comment by djordjenTT** [comment]
See this branch.
https://github.com/tenstorrent/tt-metal/commit/1634cd928bdcf51f6abfe333e1443a92f5fd57a1
Perf hit is significant but should get you unblocked. See the change description on how to run it.

**Comment by sshonTT** [comment]
@djordjenTT Hi, does your change get landed on main?
